File: tools/php/bin/views/搜索默認版本/默文檔碼→詩文.php
<?php
/*
php H:\github\Dufu-Analysis\tools\php\bin\views\搜索默認版本\默文檔碼→詩文.php 3
=> 
*/
require_once( 
	dirname( __DIR__, 3 ) . DIRECTORY_SEPARATOR .
	'lib' . DIRECTORY_SEPARATOR . '函式.php' );

check_argv( $argv, 2, 提供默文檔碼 );
$默文檔碼 = fix_doc_id( trim( $argv[ 1 ] ) );
$默認詩文檔碼 = 提取數據結構( 默認詩文檔碼 );

if( !in_array( $默文檔碼, $默認詩文檔碼 ) )
{
	echo 無頁碼, NL;
}
$詩文文檔路徑 = 默認版本詩文件夾 . $默文檔碼 . '.txt';
echo NL, file_get_contents( $詩文文檔路徑 );
?>

File: tools/python/bin/views/搜索默認版本/默文檔碼→詩文.py
'''
python H:/github/Dufu-Analysis/tools/python/bin/views/搜索默認版本/默文檔碼→詩文.py 0003
'''
# import lib
import sys
from pathlib import Path

# constant
DUFU_SHI = Path("DuFu") / "默認版本" / "詩"
current_file_path = Path(__file__).resolve()
#current_file_path = Path.cwd()
# base_path = Path(__file__).parent.resolve()
# Get the absolute path of the current file's directory
# current_dir = os.path.dirname(os.path.abspath(__file__))
base = current_file_path.parent.parent.parent.parent.parent.parent.parent

try:
    parts = [DUFU_SHI, sys.argv[1]+'.txt']
    file_path = base.joinpath(*parts)
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
        print("\n"+content)
except FileNotFoundError:
    print("Error: The file "+ file_path + " was not found.")
except UnicodeDecodeError:
    print("Error: Could not decode the file with the specified encoding. Try a different encoding.")
except IndexError:
    print("Error: 必須提供默文檔碼。")